include/nbl/video/IGPUVirtualTexture.h

```diff
@@ -17,25 +17,48 @@ namespace video
 
 namespace impl
 {
-inline core::smart_refctd_ptr<IGPUImage> createGPUImageFromCPU(IVideoDriver* _driver, asset::ICPUImage* _cpuimg)
+inline core::smart_refctd_ptr<IGPUImage> createGPUImageFromCPU(nbl::video::ILogicalDevice* logicalDevice, nbl::video::IGPUQueue* queue, asset::ICPUImage* _cpuimg)
 {
-    auto params = _cpuimg->getCreationParameters();
-    auto img = _driver->createDeviceLocalGPUImageOnDedMem(std::move(params));
+    auto cpuImageParams = _cpuimg->getCreationParameters();
+    auto gpuImage = logicalDevice->createDeviceLocalGPUImageOnDedMem(std::move(cpuImageParams));
 
     auto regions = _cpuimg->getRegions();
     assert(regions.size());
-    auto texelBuf = _driver->createFilledDeviceLocalGPUBufferOnDedMem(_cpuimg->getBuffer()->getSize(), _cpuimg->getBuffer()->getPointer());
-    _driver->copyBufferToImage(texelBuf.get(), img.get(), regions.size(), regions.begin());
 
-    return img;
+    auto gpuTexelBuffer = logicalDevice->createFilledDeviceLocalGPUBufferOnDedMem(queue, _cpuimg->getBuffer()->getSize(), _cpuimg->getBuffer()->getPointer());
+    auto gpuCommandPool = logicalDevice->createCommandPool(queue->getFamilyIndex(), IGPUCommandPool::ECF_RESET_COMMAND_BUFFER_BIT);
```

Comment:
you dont need the reset flag on the pool (its a transient pool, gets killed right away)

Reply:
btw there's a `createFilledGPUImageOnDedMem` function which you could use here, I think?

you have a problem right now, because you're going to drop your commandbuffer right after the submit (when returning from the function) and you should wait for the GPU to complete before dropping the resources used by it.

AFAIK `createFilledIGPUImage` does this for you

Reply:
function is called `createFilledDeviceLocalGPUImageOnDedMem`

Reply:
ok gonna fix it in a different way